### DISCUSSION TITLE
Frontend updates without server restart for local development of crowdsourcing task?

### DISCUSSION THREAD

**Discussion by sefinch** [facebookresearch/ParlAI#4177] [Q&A] [answered]
When locally developing the frontend for a crowdsourcing task in ParlAI, is there any way to publish the frontend changes to display in the browser without having to restart the server? 

I've found some [information](https://github.com/facebookresearch/Mephisto/tree/main/examples/static_react_task#local-development) in Mephisto about this for their example `static_react_task`, but it does not seem to be applicable to the example crowdsourcing tasks in ParlAI (e.g. the `mephisto.blueprint.link_task_source` config parameter does not exist in `model_chat` task). I'm not sure if ParlAI has some analogous documentation somewhere, but I am having difficulty finding it if it exists.

**Comment by stephenroller** [comment]
I summon @JackUrb.

**Comment by JackUrb** [comment, marked as answer]
Hi @sfillwo, @pringshia put together the `link_task_source` setup for the `static_react_tasks`. This setup is successful primarily because the static react tasks don't have any live state that needs to be restored upon a refresh (whereas ParlAI has previous chat turns, and refresh behavior wasn't always consistent). After Mephisto 1.0.0 this type of behavior should be more stable in the `ParlAIChatBlueprint` as well, so it's something that could be worth looking into. 

@pringshia any chance you can provide some direction on the steps required to enabled this for the ParlAI blueprint?

**Reply by pringshia** [reply]
Hi @sfillwo - this is the PR that added in `link_task_source` support for `static_react_task`: https://github.com/facebookresearch/Mephisto/pull/509.

The files of note are:

- `mephisto/abstractions/blueprints/static_react_task/static_react_blueprint.py`
- `mephisto/abstractions/blueprints/static_react_task/static_react_task_builder.py`

You are correct that this configuration isn't available for the `ParlAIChatBlueprint` (for the reasons Jack mentioned above - though with Mephisto 1.0 it's more reasonable to add this in).

One option here would be to mimic the implementation above and add it to `ParlAIChatBlueprint` in the [Mephisto](https://github.com/facebookresearch/Mephisto) repo. Do you feel comfortable opening up a PR with this?
